web/src/user/UserInterface.ts

```diff
@@ -364,4 +273,202 @@ export class UserInterface extends Interface {
+                type="button"
+                aria-label="${msg("Unread notifications")}"
+                @click=${onClick}
+            >
+                <span
+                    class="pf-c-notification-badge ${this.notificationsCount > 0
+                        ? "pf-m-unread"
+                        : ""}"
+                >
+                    <pf-tooltip position="top" content=${msg("Open Notification drawer")}>
+                        <i class="fas fa-bell" aria-hidden="true"></i>
+                    </pf-tooltip>
+                    <span class="pf-c-notification-badge__count">${this.notificationsCount}</span>
+                </span>
+            </button>
+        </div> `;
+    }
+
+    renderSettings() {
+        if (!this.uiConfig.enabledFeatures.settings) {
+            return nothing;
+        }
+
+        return html` <div class="pf-c-page__header-tools-item">
+            <a class="pf-c-button pf-m-plain" type="button" href="#/settings">
+                <pf-tooltip position="top" content=${msg("Settings")}>
+                    <i class="fas fa-cog" aria-hidden="true"></i>
+                </pf-tooltip>
+            </a>
+        </div>`;
+    }
+
+    renderAccessAdmin() {
+        if (!this.canAccessAdmin) {
+            return nothing;
+        }
+
+        return html`<a
+            class="pf-c-button pf-m-secondary pf-m-small pf-u-display-none pf-u-display-block-on-md"
+            href="/if/admin/"
+        >
+            ${msg("Admin interface")}
+        </a>`;
+    }
+
+    renderImpersonation() {
+        if (!this.me.original) {
+            return nothing;
+        }
+
+        const onClick = () => {
+            return new CoreApi(DEFAULT_CONFIG).coreUsersImpersonateEndRetrieve().then(() => {
+                window.location.reload();
+            });
+        };
+
+        return html`&nbsp;
+            <div class="pf-c-page__header-tools">
+                <div class="pf-c-page__header-tools-group">
+                    <ak-action-button class="pf-m-warning pf-m-small" .apiRequest=${onClick}>
+                        ${msg("Stop impersonation")}
+                    </ak-action-button>
+                </div>
+            </div>`;
+    }
+}
+
+//  ___         _
+// | _ )_  _ __(_)_ _  ___ ______
+// | _ \ || (_-< | ' \/ -_|_-<_-<
+// |___/\_,_/__/_|_||_\___/__/__/
+//
+//
+@customElement("ak-interface-user")
+export class UserInterface extends Interface {
+    @state()
+    notificationDrawerOpen = getURLParam("notificationDrawerOpen", false);
+
+    @state()
+    apiDrawerOpen = getURLParam("apiDrawerOpen", false);
+
+    ws: WebsocketClient;
+
+    @state()
+    notificationsCount = 0;
+
+    @state()
+    me?: SessionUser;
+
+    constructor() {
+        super();
+        this.ws = new WebsocketClient();
+        this.fetchConfigurationDetails();
+        configureSentry(true);
+        this.toggleNotificationDrawer = this.toggleNotificationDrawer.bind(this);
```

Comment:
All this binding is annoying, but necessary; it allows us to connect and disconnect these handlers from `window` at need.  Which would also allow us, if we wanted, to switch between "User" and "Admin" apps without actually leaving the page or reloading the environment, since we're now cleaning up after ourselves.

Reply:
Switching between all these interfaces without a full reload would definitely be very nice to have